### PR TITLE
Add PassThroughProcedure for JDBC Connector

### PR DESCRIPTION
**Pull request by harlequin**
Support `pass_through` in jdbc-base connector class. This provides the possibility to execute sql statements directly on jdbc connections.

Potentially solves issue #9163

### CODE REVIEW

**Comment by ebyhr** [comment]
Are you aware of #12315?

**Comment by findepi** [comment]
> Are you aware of #12315?

#12315 covers SELECT queries (cc @kasiafi)
this one seems to be about DML and DDL statements. (cc @wendigo @hashhar )

**Comment by harlequin** [comment]
Yes, first thoughts from my side were DML and DDL.

But we want also to use it in the mssql connector with a select and graph style syntax, for example

https://docs.microsoft.com/de-de/sql/relational-databases/graphs/sql-graph-sample?view=sql-server-ver15

**Comment by findepi** [comment]
> But we want also to use it in the mssql connector with a select and graph style syntax, for example
> 
> https://docs.microsoft.com/de-de/sql/relational-databases/graphs/sql-graph-sample?view=sql-server-ver15

Link for Enlglish-speaking people: https://docs.microsoft.com/en-us/sql/relational-databases/graphs/sql-graph-sample?view=sql-server-ver15

**Comment by harlequin** [comment]
Hey Folks,

Cla ist signed niow, how can i Trigger the Pipeline again?

**Comment by findepi** [comment]
@cla-bot check

**Comment by harlequin** [comment]
@cla-bot check

**Comment by harlequin** [comment]
Hey guys,

is there any action which I have to take care from my side, because it seems that there was something broken in general with the pipeline checks ...

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/procedure/PassThroughProcedure.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    }
+
+    public void passThrough(ConnectorSession session, ConnectorAccessControl accessControl, String statement)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(getClass().getClassLoader())) {
+            doPassThrough(session, accessControl, statement);
+        }
+    }
+
+    public void doPassThrough(ConnectorSession session, ConnectorAccessControl accessControl, String statement)
```
Comment (findepi) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/procedure/PassThroughProcedure.java:76]: `accessControl` is unused

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/procedure/PassThroughProcedure.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    {
+        try (Connection connection = this.jdbcClient.getConnection(session, new JdbcSplit(Optional.empty()))) {
```
Comment (findepi) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/procedure/PassThroughProcedure.java:78]: there is no split in action, don't create a fake one

File: docs/src/main/sphinx/connector/jdbc-procedures.fragment [hunk 3]
```diff
@@ -10,3 +10,13 @@ Procedures
 
       USE example.myschema;
       CALL system.flush_metadata_cache();
+
+* ``system.pass_through('statement')``
```
Comment (findepi) [inline docs/src/main/sphinx/connector/jdbc-procedures.fragment:14]: we have `system.query`  table function.

what should be most consistent name for this?

Comment (harlequin) [reply docs/src/main/sphinx/connector/jdbc-procedures.fragment:14]: Would it make sense to rename it also to `query`, or would it confuse users that there is a `DDL query` and a `table query`?

Can it even be the same name?

Comment (findepi) [reply docs/src/main/sphinx/connector/jdbc-procedures.fragment:14]: > Can it even be the same name?

i think the procedures and table functions do not share a namespace today, but i don't know what's the long term intention. cc @martint @dain @kasiafi

Comment (kokosing) [reply docs/src/main/sphinx/connector/jdbc-procedures.fragment:14]: how about `system.update` to match JDBC update statement, and that table function should return number of affected rows.

**Comment by neerajvash8** [comment]
Hey guys, we are really forward for this functionality. Could this be merged if it satisfy the requirement?

**Comment by mosabua** [comment]
:wave: @harlequin  - this PR has become inactive. We hope you are still interested in working on it. Please let us know, and we can try to get reviewers to help with that.

Also @findepi @kokosing please chime with other info about needed changes or if this work is no longer needed.

We're working on closing out old and inactive PRs, so if you're too busy or this has too many merge conflicts to be worth picking back up, we'll be making another pass to close it out in a few weeks.

**Comment by ebyhr** [comment]
Superseded by #22556

**Pull request closed without merging**